contracts/contracts/UserReplicaSetManager.sol

```diff
@@ -0,0 +1,120 @@
+    }
+
+    // Current userId to replica set
+    mapping (uint => ReplicaSet) artistReplicaSets;
+
+    // TODO: Capture deployer here
+    constructor(
+        address _registryAddress,
+        bytes32 _userFactoryRegistryKey,
+        uint _networkId
+    ) SigningLogic("User Replica Set Manager", "1", _networkId) public
+    {
+        require(
+            _registryAddress != address(0x00) &&
+            _userFactoryRegistryKey.length != 0,
+            "requires non-zero _registryAddress"
+        );
+
+        registry = RegistryInterface(_registryAddress);
+        userFactoryRegistryKey = _userFactoryRegistryKey;
+        deployer = msg.sender;
+    }
+
+    // Called from dataOwnerWallet value as msg.sender
+    function registerCreatorNode(uint _spID) external {
+        // TODO: Handle updates
+        require(spIdToCreatorNode[_spID] == address(0x0), "No value permitted prior to setting spID");
+        spIdToCreatorNode[_spID] = msg.sender;
+    }
+
+    // Function used to permission updates to a given user's replica set 
+    function updateReplicaSet(
+        uint _userId,
+        uint _primary,
+        uint[] calldata _secondaries,
+        uint _oldPrimary,
+        uint[] calldata _oldSecondaries) external
+      {
+          // Caller's notion of existing primary must match regisered value on chain
+          require(artistReplicaSets[_userId].primary == _oldPrimary,  "Invalid prior configuration"); 
+
+          // Get user object from UserFactory
+          UserFactoryInterface userFactory = UserFactoryInterface(registry.getContract(userFactoryRegistryKey));
+          require(userFactory.userExists(_userId), "Valid user required");
+
+          (address userWallet, ) = userFactory.getUser(_userId);
+
+          // A valid updater can be one of the dataOwnerWallet, existing creator node, or contract deployer
+          bool validUpdater = false;
+          // TODO: Replace msg.sender below with recovered signature from _subjectSig object
+          if (msg.sender == userWallet || msg.sender == spIdToCreatorNode[_oldPrimary] || msg.sender == deployer) {
+              validUpdater = true;
+          }
+          // Caller's notion of secondary values must match registered value on chain
+          // A secondary node can also be considered a valid updater
+          for (uint i = 0; i < _oldSecondaries.length; i++) {
+              require(
+                  artistReplicaSets[_userId].secondaries[i] == _oldSecondaries[i],
+                  "Invalid prior secondary configuration"
+              );
+
+              // A valid updater has been found
+              if (msg.sender == spIdToCreatorNode[_oldSecondaries[i]]) {
+                validUpdater = true;
+              }
+          }
+          require(validUpdater == true, "Invalid update operation");
+
+          // Confirm primary and every incoming secondary is valid
+          require(spIdToCreatorNode[_primary] != address(0x00), "Primary must exist"); 
```

Comment:
considering a removal of this entirely...as long as the sender is valid then shouldn't setting null values be allowed?